### PR TITLE
eww: 0.5.0 -> 0.6.0; other cleanup

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

https://github.com/elkowar/eww/releases/tag/v0.6.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by w-lfchen** [comment]
I'd love to join as a maintainer.
If you want to add me, my handle is `w-lfchen`.
Or should I open a separate pr for that?

**Comment by eclairevoyant** [comment]
@w-lfchen added, thanks for mentioning.

**Comment by w-lfchen** [comment]
Result of `nixpkgs-review pr 305784` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eww</li>
  </ul>
</details>

**Review by w-lfchen — APPROVED** [review]
running `eww -V` will return `0.5.0`, this is due to the cargo version being out of date for the 0.6.0 release tag, it got updated in https://github.com/elkowar/eww/commit/8397744f403c87ab39523b4bf9e0216ad444f1bb
The new functionality is present.

File: pkgs/by-name/ew/eww/package.nix [hunk 1]
```diff
@@ -7,24 +7,30 @@
...
   pname = "eww";
-  version = "0.5.0";
+  version = "0.6.0";
 
   src = fetchFromGitHub {
     owner = "elkowar";
     repo = "eww";
-    rev = "v${version}";
-    hash = "sha256-HBBz1NDtj2TnDK5ghDLRrAOwHXDZlzclvVscYnmKGck=";
+    rev = "refs/tags/v${version}";
```
Comment (krovuxdev) [inline pkgs/by-name/ew/eww/package.nix:20]: pls remove `refs/tags/` because it already has the tags from the eww repository.

Comment (eclairevoyant) [reply pkgs/by-name/ew/eww/package.nix:20]: There's really no reason to remove it; this ensures that we're explicitly fetching a tag and not an identically-named branch, or something else that can confuse an update script or update bot.

Comment (krovuxdev) [reply pkgs/by-name/ew/eww/package.nix:20]: with `refs/tags` will work but also without `refs/tags` the same, but as you saw the previous eww commits did not have that `refs/tags` and it works fine. 
so please. if your reason with `refs/tags` is ok.

**Pull request merged**